addons/l10n_in_pos/static/src/overrides/models/pos_order_line.js

```diff
@@ -10,7 +10,7 @@ patch(PosOrderline.prototype, {
     getDisplayData() {
         return {
             ...super.getDisplayData(),
-            l10n_in_hsn_code: this.get_product().l10n_in_hsn_code,
+            l10n_in_hsn_code: this.get_product().l10n_in_hsn_code || "",
```

Comment:
If it wasn't a `String` what was it ? Why are you fixing it like this ?

Reply:
@rugo-odoo It was undefined. The props type is correct. It's an optional string field. However, upon validation, if it exists, it must be a string. Therefore, it must either be missing or be a string.